cocos/physics/bullet/constraints/bullet-fixed-constraint.ts

```diff
@@ -0,0 +1,100 @@
+/*
+ Copyright (c) 2020 Xiamen Yaji Software Co., Ltd.
+
+ https://www.cocos.com/
+
+ Permission is hereby granted, free of charge, to any person obtaining a copy
+ of this software and associated engine source code (the "Software"), a limited,
+ worldwide, royalty-free, non-assignable, revocable and non-exclusive license
+ to use Cocos Creator solely to develop games on your target platforms. You shall
+ not use Cocos Creator software for developing other software or tools that's
+ used for developing games. You are not granted to publish, distribute,
+ sublicense, and/or sell copies of Cocos Creator.
+
+ The software or tools in this License Agreement are licensed, not sold.
+ Xiamen Yaji Software Co., Ltd. reserves all rights not expressly granted to you.
+
+ THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ THE SOFTWARE.
+ */
+
+/* eslint-disable new-cap */
+import { BulletConstraint } from './bullet-constraint';
+import { IFixedConstraint } from '../../spec/i-physics-constraint';
+import { IVec3Like, Quat, Vec3, Mat4 } from '../../../core';
+import { FixedConstraint, HingeConstraint, PhysicsSystem } from '../../framework';
+import { BulletRigidBody } from '../bullet-rigid-body';
+import { BulletCache, CC_MAT4_0, CC_QUAT_0, CC_V3_0 } from '../bullet-cache';
+import { bt } from '../instantiated';
+import { bullet2CocosVec3, cocos2BulletQuat, cocos2BulletVec3 } from '../bullet-utils';
+import { BulletWorld } from '../bullet-world';
+
+export class BulletFixedConstraint extends BulletConstraint implements IFixedConstraint {
+    setBreakForce (v: number): void {
+        bt.TypedConstraint_setMaxImpulseThreshold(this._impl, v);
+    }
+
+    setBreakTorque (v: number): void {
+        // not supported
+    }
+
+    get constraint (): FixedConstraint {
+        return this._com as FixedConstraint;
+    }
+
+    onComponentSet () {
```

Comment:
this implementation seems a little different fro  another two constraint implementations